telemetry-core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java

```diff
@@ -71,24 +71,25 @@ private String buildUserAgent(String additionalUserAgent) {
     return BASE_USER_AGENT_VALUE + " " + additionalUserAgent;
   }
 
-  public Response send(String json, UUID batchId)
+  public Response send(String json, UUID batchId, String batchType)
```

Comment:
Can we add Javadoc to this public method? Also, is there concern about changing the arguments of a public API? Technically, this is a breaking change.

Reply:
That's a good point. I suppose we could add a new overloaded method or something to maintain backwards compatibility though that wouldn't force users to use the new API that adds the improved logging. Also, adding a new method just to get different logging output is not great now that I think about.

That said, given that this is largely an internal SDK used by us to build exporters I think we should have some leeway to make breaking changes as long as they're documented in the release notes.

Reply:
Any insight why its a public method instead of package private?

Reply:
Not really, unfortunately.

Reply:
I'll need to add a javadoc and also update the changelog